### ISSUE TITLE
2.19.5 Failed to parse @flow argument

### ISSUE THREAD

**Issue by smeyanoff** [PrefectHQ/prefect#14025]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Prefect documentation for this issue.
- [X] I checked that this issue is related to Prefect and not one of its dependencies.

### Bug summary

I'm trying "prefect deploy --all" without installing all environment, only prefect. 
I call my flow with "name" parameter, but it isn't work.

### Reproduction

```python3
flow.py

...
@flow(
    name=(
        f'{os.getenv("DEPLOYMENT_NAME")}'
        f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'
    ),
    log_prints=True,
)
def spark_kubernetes_flow(
    script_path: str,
    config_path: str,
    object_name: str,
    prefect_deployment_name: str,
) -> None:
    upload_script_to_s3(script_path, object_name)
    create_spark_application(object_name, config_path, prefect_deployment_name)
    monitor_spark_application(prefect_deployment_name)


prefect.yaml

# Generic metadata about this project
name: '{{ $DEPLOYMENT_NAME }}'
prefect-version: '{{ $PREFECT_VERSION }}'
# build section allows you to manage and build docker images
build:
  - prefect_docker.deployments.steps.build_docker_image:
      id: build-image
      requires: prefect-docker
      image_name: '{{ $REGISTRY }}/{{ $DEPLOYMENT_NAME }}'
      tag: '{{ $DEPLOYMENT_TAG }}'
      dockerfile: Dockerfile
push:
  - prefect_docker.deployments.steps.push_docker_image:
      requires: prefect-docker
      image_name: '{{ build-image.image_name }}'
      tag: '{{ build-image.tag }}'
pull:
  - prefect.deployments.steps.set_working_directory:
      directory: /opt/prefect
# the definitions section allows you to define reusable components for your deployments
definitions:
  tags: &common_tags [spark, '{{ $PREFECT_WORK_POOL_NAME }}']
  work_pool: &common_work_pool
    name: '{{ $PREFECT_WORK_POOL_NAME }}'
    work_queue_name: default
    job_variables:
      image: '{{ build-image.image }}'
  parameters: &parameters
    prefect_deployment_name: '{{ $DEPLOYMENT_NAME }}'
    script_path: /application/{{ $SPARK_FILE_NAME }}
    config_path: /application/spark_application.yaml
    object_name: '{{ $PREFECT_WORK_POOL_NAME }}/{{ $DEPLOYMENT_NAME }}/{{ build-image.tag}}.py'
# the deployments section allows you to provide configuration for deploying flows
deployments:
  - name: '{{ $DEPLOYMENT_NAME }}'
    version: '{{ build-image.tag }}'
    description:
    tags: *common_tags
    work_pool: *common_work_pool
      # flow-specific fields
    entrypoint: flows/flow.py:spark_kubernetes_flow
    parameters: *parameters

pyproject.toml
[tool.poetry]
name = "spark"
version = "2.0.4"
description = ""
authors = ["Nesmeyanov Dmitry <nesmeyanov.dv@avgd.pro>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.10"
kubernetes = "<30.0.0"
boto3 = "^1.34.121"
pyyaml = "^6.0.1"

[tool.poetry.group.lint.dependencies]
isort = "^5.12.0"
black = "^24.3.0"
flake8 = "^6.1.0"
flake8-annotations = "^3.0.1"
mypy = "^1.7.0"
pylint = "^3.0.2"

[tool.poetry.group.addons.dependencies]
pre-commit = "^3.7.1"

[tool.poetry.group.build.dependencies]
requests = "<2.32.0"
prefect = "^2.19.5"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"


[tool.black]
line-length = 80

command
yes N | poetry run prefect deploy --all
```

### Error

```python3
yes N | poetry run prefect deploy --all
/root/.cache/pypoetry/virtualenvs/spark-EXuGVG_b-py3.10/lib/python3.10/site-packages/prefect/utilities/importtools.py:437: RuntimeWarning: coroutine 'submit_autonomous_task_run_to_engine' was never awaited
  logger.debug("Failed to compile: %s", e)
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
12:42:58.243 | INFO    | prefect.flows - Failed to parse @flow argument: `name=f'{os.getenv("DEPLOYMENT_NAME")}'
        f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'` due to the following error. Ignoring and falling back to default behavior.
Traceback (most recent call last):
  File "/root/.cache/pypoetry/virtualenvs/spark-EXuGVG_b-py3.10/lib/python3.10/site-packages/prefect/flows.py", line 1865, in load_flow_argument_from_entrypoint
    evaluated_value = eval(literal_arg_value, namespace)  # type: ignore
  File "<string>", line 2
    f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'
IndentationError: unexpected indent
Running deployment build steps...
 > Running build_docker_image step...
Collecting prefect-docker
...
```

### Versions

```Text
*installing only *build* group
`poetry install --sync --no-root --only build`

[tool.poetry]
name = "spark"
version = "2.0.4"
description = ""
authors = ["Nesmeyanov Dmitry <nesmeyanov.dv@avgd.pro>"]
readme = "README.md"

[tool.poetry.dependencies]
python = "^3.10"
kubernetes = "<30.0.0"
boto3 = "^1.34.121"
pyyaml = "^6.0.1"

[tool.poetry.group.lint.dependencies]
isort = "^5.12.0"
black = "^24.3.0"
flake8 = "^6.1.0"
flake8-annotations = "^3.0.1"
mypy = "^1.7.0"
pylint = "^3.0.2"

[tool.poetry.group.addons.dependencies]
pre-commit = "^3.7.1"

[tool.poetry.group.build.dependencies]
requests = "<2.32.0"
prefect = "^2.19.5"

[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"


[tool.black]
line-length = 80
```

**Comment by jakekaplan** [comment]
Hi @smeyanoff thanks for filing this. It seems my simplified reproduction in https://github.com/PrefectHQ/prefect/issues/13895 masked another issue that your code had that my reproduction didn't. The issue seems to be with the multiline f-string. While we resolve this if you do a single line f-string it should work:
`name=f'{os.getenv("DEPLOYMENT_NAME")}-{os.getenv("PREFECT_WORK_POOL_NAME")}'`


### Reproduction:

```python
# my_flow.py
import os
from prefect import flow


@flow(
    name=(
        f'{os.getenv("DEPLOYMENT_NAME")}'
        f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'
    ),
    log_prints=True,
)
def my_flow():
    pass
```

```
prefect deploy my_flow.py:my_flow
```

```
09:25:17.734 | INFO    | prefect.flows - Failed to parse @flow argument: `name=f'{os.getenv("DEPLOYMENT_NAME")}'
        f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'` due to the following error. Ignoring and falling back to default behavior.
Traceback (most recent call last):
  File "/opt/miniconda3/envs/demo-flows310/lib/python3.10/site-packages/prefect/flows.py", line 1865, in load_flow_argument_from_entrypoint
    evaluated_value = eval(literal_arg_value, namespace)  # type: ignore
  File "<string>", line 2
    f'-{os.getenv("PREFECT_WORK_POOL_NAME")}'
IndentationError: unexpected indent
```
